### PR TITLE
Widget list and explanation

### PR DESCRIPTION
**Pull request by Tolsa**
- add available widgets and described the one we know how to describe, a few are not listed, but we can't really picture what they do (I at least can't...)
- add picture to describe some of them

### CODE REVIEW

File: _includes/express-mongoose/fields.md [hunk 1]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+
+For reach field you have you have several display options called "Widgets". You may find these options in your collection settings under the fields section.
+
+(first picture here)
+
+Depending on the field type in your DB, you have different choices of Widgets available :
+
+| Type                  | Widget                                                                                           | Description
+| --------------------- | ---------------------------------------------------------------------------------------------- | ---------------------------------------------
+| string           | link | clickable link instead of a raw string
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display a link.

File: _includes/express-mongoose/fields.md [hunk 2]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | text area | larger area allowed for the string than without widget
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Enlarge the text area.

File: _includes/express-mongoose/fields.md [hunk 3]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | color picker | change the field to a color picker (hexadecimal code)
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Change the field to a color picker (hexadecimal code).

File: _includes/express-mongoose/fields.md [hunk 4]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | image viewer | it displays the image instead of the raw url of it
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display an image instead of a URL link.

File: _includes/express-mongoose/fields.md [hunk 5]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | file picker | replace the field by a button to add a file from your computer
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Add a file from your computer.

File: _includes/express-mongoose/fields.md [hunk 6]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | rich text editor | larger area to write text with many formatting options associated
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Write formatted text.

File: _includes/express-mongoose/fields.md [hunk 7]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | JSON editor | allow you to respect the JSON format with proper indentation when you open a record. Important note : on a tabular view, you will get the default [object, Object] indication and not the full JSON. The reason for this is that JSON lengths are very variable and display them fully while keeping a convenient UI is not possible in most cases
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: @Tolsa [object, Object] is a bug on our side. We'll fix it asap so no need to specify it here.

Display a JSON field.

File: _includes/express-mongoose/fields.md [hunk 8]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | dropdown | for strings such as 'blue, yellow, red', transform the string into a dropdown of items. There will as many items as words separated by ','
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Put a drop-down menu of fields among pre-defined options.

File: _includes/express-mongoose/fields.md [hunk 9]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | document viewer | display your document as in a preview window instead of just the raw URL
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display your document in a preview window.

File: _includes/express-mongoose/fields.md [hunk 10]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | map | display a location on a map if your string is made of coordinates respecting the following format 'lat, long'. It matches the Google maps default formatting if you want to pick coordinates from there
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display a location on a map (⚠️ your field must be a "string" respecting the following format "lat, long").

File: _includes/express-mongoose/fields.md [hunk 11]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | address | implement an auto-complete with addresses on the field
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Allow auto-complete on an address field.

File: _includes/express-mongoose/fields.md [hunk 12]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+| date             | text input | convert your date into string
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Convert your date into a string.

File: _includes/express-mongoose/fields.md [hunk 13]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | date picker | allow you to pick a date
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Use the date picker to select a date.

File: _includes/express-mongoose/fields.md [hunk 14]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+| number/integer   | belongsTo select | displays a dropdown list of related data
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display a dropdown list of related data.

File: _includes/express-mongoose/fields.md [hunk 15]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+|                  | price | compute your number in the currency you desire
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Compute your number in the currency of your choice.

File: _includes/express-mongoose/fields.md [hunk 16]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+| array            | carrousel | display an array of image url as a carrousel
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: Display an array of image URL as a carousel.

File: _includes/express-mongoose/fields.md [hunk 17]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
...
+{:.markdown-table}
+
+<br/>
+
+You can see below some of the widget and how they look:
```
Comment (louisteenbrink) [inline _includes/express-mongoose/fields.md]: some of the widget(s)?

File: _includes/express-mongoose/fields.md [hunk 18]
```diff
@@ -184,3 +184,37 @@ Liana.collection('users', {
 Sometimes, searching on all fields is not relevant and may even involve big performance issues. You can restrict the search on specific fields only using the option `searchFields`, which should be defined in your `forest/` folder. In this example, we configure Forest to only search on the fields `name` and `industry` of our collection `Company`.
 
 <div data-github-gist="Tolsa/b60b1e080c84a7ea0118a64868c891e8"></div>
+
+## Available Widget Options
+
+For reach field you have you have several display options called "Widgets". You may find these options in your collection settings under the fields section.
```
[the remark this answers is not in the document]
Comment (louisteenbrink) [reply _includes/express-mongoose/fields.md]: @Tolsa @SeyZ 
⚠️ Lot of typos, grammar or vocabulary errors in the PR documentation.

We have to be very, very, very rigorous on the documentation and limit our content debt as much as possible.

2 solutions:
- @Tolsa, please use tools like those listed above. I am not a master in this field, but these tools help a lot.
- Ask maybe a native guy in the team (Xavier or Jacob) before submitting the PR.

When submitting a request, it must be almost perfect. The review should not be mainly dedicated to vocabulary or grammar correction. But more on consistency.

[the remark this answers is not in the document]
Comment (Tolsa) [reply _includes/express-mongoose/fields.md]: Completely my bad on this @louisteenbrink , I wanted to have a first opinion on this to be sure the format was OK for you, I wrote the wording too quickly knowing that I might have to change the structure eventually.

That being said, I agree that I shouldn't have made a PR this early, if I want to get feedback I better show it to you directly at the office ;).

**Pull request merged**